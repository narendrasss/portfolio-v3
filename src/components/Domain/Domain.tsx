import React from 'react';
import { TextProps } from 'rebass';
import { Text } from '../base/Text';

export interface DomainProps extends TextProps {
  url: string;
}

const Domain: React.FC<DomainProps> = ({ url, ...props }) => {
  const [name, domain] = url.split('.');
  return (
    <Text fontWeight={600} {...props}>
      {name}
      <Text as="span" fontWeight={300}>
        .{domain}
      </Text>
    </Text>
  );
};

export default Domain;
