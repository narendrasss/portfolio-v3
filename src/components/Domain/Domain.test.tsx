import React from 'react';
import { render } from 'react-testing-library';
import Domain from './Domain';

describe('Domain', () => {
  const url = 'narendras.io';

  it('renders without crashing', () => {
    const { asFragment } = render(<Domain url={url} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders own domain', () => {
    const ownDomain = url.split('.')[0];
    const { queryByText } = render(<Domain url={url} />);
    expect(queryByText(ownDomain)).toBeTruthy();
  });

  it('renders top-level domain', () => {
    const topLevelDomain = url.split('.')[1];
    const { queryByText } = render(<Domain url={url} />);
    expect(queryByText('.' + topLevelDomain)).toBeTruthy();
  });
});
