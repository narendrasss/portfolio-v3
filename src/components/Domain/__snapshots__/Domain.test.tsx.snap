// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Domain renders without crashing 1`] = `
<DocumentFragment>
  <p
    class="sc-bdVaJa sc-htpNat jzCLQO"
    font-family="sans"
    font-size="0,1"
    font-weight="600"
  >
    narendras
    <span
      class="sc-bdVaJa sc-htpNat fYauba"
      font-family="sans"
      font-size="0,1"
      font-weight="300"
    >
      .io
    </span>
  </p>
</DocumentFragment>
`;
